controllers/etcd_controller.go

```diff
@@ -840,31 +612,48 @@ func (r *EtcdReconciler) reconcileEtcd(ctx context.Context, logger logr.Logger,
 		return nil, nil, fmt.Errorf("Spec.Replicas should not be even number: %d", etcd.Spec.Replicas)
 	}
 
-	val := componentetcd.Values{
-		ConfigMap: componentconfigmap.GenerateValues(etcd),
-		Lease:     componentlease.GenerateValues(etcd),
-		Service:   componentservice.GenerateValues(etcd),
+	etcdImage, etcdBackupImage, err := getEtcdImages(r.ImageVector, etcd)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	leaseDeployer := componentlease.New(r.Client, etcd.Namespace, val.Lease)
+	if etcd.Spec.Etcd.Image == nil {
+		if etcdImage == "" {
+			return nil, nil, fmt.Errorf("either etcd resource or image vector should have %s image while deploying statefulset", common.Etcd)
+		}
+	} else {
+		etcdImage = *etcd.Spec.Etcd.Image
+	}
+
+	if etcd.Spec.Backup.Image == nil {
+		if etcdBackupImage == "" {
+			return nil, nil, fmt.Errorf("either etcd resource or image vector should have %s image while deploying statefulset", common.BackupRestore)
```

Comment:
This will result in re-queueing of request and since the image is now nil this will keep getting re-queued right?